float64 leg_1
float64 leg_2

time stamp